src/test/regress/expected/foreign_key_restriction_enforcement.out

```diff
@@ -0,0 +1,1089 @@
+DEBUG:  building index "on_update_fkey_table_pkey" on table "on_update_fkey_table"
+ROLLBACK;
+-----
+--- Now, start testing the other way araound
+-----
+-- case 4.1: SELECT to a dist table is follwed by a SELECT to a reference table
+BEGIN;
+	SELECT count(*) FROM on_update_fkey_table WHERE value_1 = 99;
+ count 
+-------
+    10
+(1 row)
+
+	SELECT count(*) FROM referece_table;
+ count 
+-------
+   101
+(1 row)
+
+ROLLBACK;
+-- case 4.2: SELECT to a dist table is follwed by a DML to a reference table
+BEGIN;
+	SELECT count(*) FROM on_update_fkey_table WHERE value_1 = 99;
+ count 
+-------
+    10
+(1 row)
+
+	UPDATE referece_table SET id = 101 WHERE id = 99;
+DEBUG:  switching to sequential query execution mode
+DETAIL:  Reference relation "referece_table" is modified, which might have cascaded to hash distributed relations via foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
+ERROR:  attempt to set query execution mode to sequential failed since a parallel query has already been executed in the same transaction
+HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
+ROLLBACK;
+-- case 4.3: SELECT to a dist table is follwed by an unrelated DDL to a reference table
+BEGIN;
+	SELECT count(*) FROM on_update_fkey_table WHERE value_1 = 99;
+ count 
+-------
+    10
+(1 row)
+
+	ALTER TABLE referece_table ADD COLUMN X INT;
+ERROR:  cannot execute "DDL" on relation "referece_table" since it holds conflicting locks on relation "on_update_fkey_table" due to a previous parallel "SELECT" access in the same transaction 
+HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
+ROLLBACK;
+-- case 4.4: SELECT to a dist table is follwed by a DDL to a reference table
+BEGIN;
+	SELECT count(*) FROM on_update_fkey_table WHERE value_1 = 99;
+ count 
+-------
+    10
+(1 row)
+
+	ALTER TABLE referece_table ALTER COLUMN id SET DATA TYPE smallint;
+DEBUG:  rewriting table "referece_table"
+DEBUG:  building index "referece_table_pkey" on table "referece_table"
+DEBUG:  validating foreign key constraint "fkey"
+ERROR:  cannot execute "DDL" on relation "referece_table" since it holds conflicting locks on relation "on_update_fkey_table" due to a previous parallel "SELECT" access in the same transaction 
+HINT:  Try re-running the transaction with "SET LOCAL citus.multi_shard_modify_mode TO 'sequential';"
+ROLLBACK;
+-- case 4.5: SELECT to a dist table is follwed by a TRUNCATE
+BEGIN;
+	SELECT count(*) FROM on_update_fkey_table WHERE value_1 = 99;
+ count 
+-------
+    10
+(1 row)
+
+	TRUNCATE referece_table CASCADE;
+NOTICE:  truncate cascades to table "on_update_fkey_table"
+ERROR:  cannot execute "DDL" on relation "referece_table" since it holds conflicting locks on relation "on_update_fkey_table" due to a previous parallel "SELECT" access in the same transaction 
```

Comment:
> since it holds conflicting locks on relation "on_update_fkey_table" due to a previous parallel "SELECT" access in the same transaction

the conflicting locks bit is probably not going to be meaningful to most users (and it's not necessarily true), maybe:

"because there was a parallel SELECT on "on_update_fkey_table" in the same transaction"